File: SurveyManagementService/models/conductor.go
package models

import (
	"time"
)

type ConductorType string

const (
	ConductorTypeInstitute ConductorType = "INSTITUTE"
	ConductorTypeCompany   ConductorType = "COMPANY"
)

type Conductor struct {
	ConductorID   uint         `json:"id" gorm:"primaryKey"`
	UserID        uint         `json:"user_id" gorm:"unique"`
	Name          string       `json:"name"`
	Type          ConductorType `json:"type" gorm:"type:varchar(20)"`
	Description   string       `json:"description"`
	ContactEmail  string       `json:"contact_email"`
	ContactPhone  string       `json:"contact_phone"`
	Address       string       `json:"address"`
	OfficialEmail string       `json:"official_email" gorm:"unique"` // Institute/Company email for verification
	IsVerified    bool         `json:"is_verified" gorm:"default:false"`
	VerificationCode string    `json:"verification_code,omitempty" gorm:"-"`
	CreatedAt     time.Time    `json:"created_at"`
	UpdatedAt     time.Time    `json:"updated_at"`
}



type ConductorRegistrationRequest struct {
	Name          string        `json:"name"`
	Type          ConductorType `json:"type"` // Enum: INDIVIDUAL, INSTITUTE, COMPANY
	Description   string        `json:"description"`
	ContactEmail  string        `json:"contact_email"`
	ContactPhone  string        `json:"contact_phone"`
	Address       string        `json:"address"`
	OfficialEmail string        `json:"official_email"`
}

type ConductorUpdateRequest struct {
	Name          string        `json:"name"`
	Description   string        `json:"description"`
	ContactEmail  string        `json:"contact_email"`
	ContactPhone  string        `json:"contact_phone"`
	Address       string        `json:"address"`
}

File: AuthService/Models/Conductor.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace AuthService.Models
{
    public class Conductor
    {
        public int ConductorId { get; set; }
        public int UserId { get; set; }

        [JsonIgnore]
        public User User { get; set; }
        public string Name { get; set; }
        
        public ConductorType ConductorType { get; set; }
        public string Description { get; set; }
        public string ContactEmail { get; set; }
        public string ContactPhone { get; set; }
        public string Address { get; set; }
        public bool IsVerified { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }

    public enum ConductorType
    {
        INDIVIDUAL,
        INSTITUTE,
        COMPANY
    }
}